src/mono/IosAppBuilderTasks/IosAppBuilderTask.cs

```diff
@@ -0,0 +1,143 @@
+{
+    /// <summary>
+    /// Path to arm64 AOT cross-compiler (mono-aot-cross)
+    /// It's not used for x64 (Simulator)
+    /// </summary>
+    public string? CrossCompiler { get; set; }
+
+    /// <summary>
+    /// ProjectName is used as an app name, bundleId and xcode project name
+    /// </summary>
+    [Required]
+    public string ProjectName { get; set; } = ""!;
+
+    /// <summary>
+    /// Target directory with *dll and other content to be AOT'd and/or bundled
+    /// </summary>
+    [Required]
+    public string AppDir { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to Mono public headers (*.h)
+    /// </summary>
+    [Required]
+    public string MonoInclude { get; set; } = ""!;
+
+    /// <summary>
+    /// This library will be used as an entry-point (e.g. TestRunner.dll)
+    /// </summary>
+    [Required]
+    public string EntryPointLib { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to a custom main.m with custom UI
+    /// A default one is used if it's not set
+    /// </summary>
+    public string? NativeMainSource { get; set; }
+
+    /// <summary>
+    /// Produce optimized binaries (e.g. use -O2 in AOT)
+    /// </summary>
+    public bool Optimized { get; set; }
+
+    /// <summary>
+    /// Target arch, can be "arm64" (device) or "x64" (simulator) at the moment
+    /// </summary>
+    [Required]
+    public string Arch { get; set; } = ""!;
+
+    /// <summary>
+    /// DEVELOPER_TEAM provisioning, needed for arm64 builds.
+    /// </summary>
+    public string? DevTeamProvisioning { get; set; }
+
+    /// <summary>
+    /// Only generate an xcode project
+    /// </summary>
+    public bool GenerateOnlyXcodeProject { get; set; }
+
+    /// <summary>
+    /// Path to *.app bundle
+    /// </summary>
+    [Output]
+    public string AppBundlePath { get; set; } = ""!;
+
+    /// <summary>
+    /// Path to xcode project
+    /// </summary>
+    [Output]
+    public string XcodeProjectPath { get; set; } = ""!;
+
+    public override bool Execute()
+    {
+        Utils.Logger = Log;
+        bool isDevice = Arch.Equals("arm64", StringComparison.InvariantCultureIgnoreCase);
+        if (isDevice && string.IsNullOrEmpty(CrossCompiler))
+        {
+            throw new ArgumentException("arm64 arch requires CrossCompiler");
+        }
+
+        // escape spaces
+        ProjectName = ProjectName.Replace(" ", "-");
+
+        var stopWatch = Stopwatch.StartNew();
+
+        // see https://github.com/dotnet/runtime/issues/34448
+        // Can be exposed as an input argument if needed
+        string[] excludes = {"System.Runtime.WindowsRuntime.dll"};
+        string[] libsToAot = Directory.GetFiles(AppDir, "*.dll")
+            .Where(f => !excludes.Contains(Path.GetFileName(f)))
+            .ToArray();
+
+        string binDir = Path.Combine(AppDir, $"bin-{ProjectName}-{Arch}");
+        Directory.CreateDirectory(binDir);
+
+        // run AOT compilation only for devices
+        if (isDevice)
```

Comment:
Why not make it configurable?

Reply:
because AOT won't work for simulator as is (And JIT won't work for device).
I **can** change AOTCompiler to support simulator as well but doesn't make sense? In mono/mono/sdks/ios we use JIT for simulator for ci